### PR TITLE
[pulsar-broker] Introduce publish rate-limiting on topic

### PR DESCRIPTION
**Pull request by rdhabalia**
### Motivation
It addresses  #3985 by introducing publish rate-limiting on the topic.

### CODE REVIEW

**Comment by rdhabalia** [comment]
retest this please

**Comment by rdhabalia** [comment]
rerun java8 tests
rerun integration tests

**Comment by sijie** [comment]
do we need this for 2.4.0 or shall we move it to 2.5.0?

**Comment by sijie** [comment]
@rdhabalia I moved this feature to 2.5.0. If it is needed in 2.4.0, please move it back :-)

**Comment by rdhabalia** [comment]
rerun java8 tests

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/Namespaces.java [hunk 1]
```diff
@@ -901,6 +902,27 @@ void setBookieAffinityGroup(String namespace, BookieAffinityGroupData bookieAffi
      */
     void splitNamespaceBundle(String namespace, String bundle, boolean unloadSplitBundles) throws PulsarAdminException;
 
+    /**
+     * Set message-publish-rate (topics under this namespace can dispatch this many messages per second)
+     *
+     * @param namespace
+     * @param dispatchRate
```
Comment (aahmed-se) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/Namespaces.java]: change dispatchRate to publishMsgRate

**Comment by rdhabalia** [comment]
@merlimat can you also review this PR before we merge? we have need of this feature so, want to merge if possible.

**Comment by rdhabalia** [comment]
rerun java8 tests

**Comment by rdhabalia** [comment]
rerun java8 tests

**Comment by rdhabalia** [comment]
rerun java8 tests

**Comment by rdhabalia** [comment]
rerun java8 tests
rerun integration tests

**Comment by wolfstudy** [comment]
Change the Milestone to 2.4.2, because of conflict.

**Pull request merged**